xen-bugtool

```diff
@@ -723,8 +723,9 @@ def collect_data(subdir, archive):
             try:
                 s = no_unicode(v["func"](cap))
             except Exception:
-                s = traceback.format_exc()
-                log(s)
+                backtrace = traceback.format_exc()  # type: str
+                log(backtrace)
+                s = backtrace.encode()
```

Comment:
This is the fix: It needs to convert the backtrace to bytes for the `BytesIO` that is used by `tarfile` and `zipfile` to include the data in the output archive.